src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/DirectoryException.cs

```diff
@@ -38,22 +39,37 @@ public class DirectoryOperationException : DirectoryException, ISerializable
 
         public DirectoryOperationException(string message, Exception inner) : base(message, inner) { }
 
-        public DirectoryOperationException(DirectoryResponse response) : base(SR.DefaultOperationsError)
+        public DirectoryOperationException(DirectoryResponse response) :
+            base(CreateMessage(response, message: null))
         {
             Response = response;
         }
 
-        public DirectoryOperationException(DirectoryResponse response, string message) : base(message)
+        public DirectoryOperationException(DirectoryResponse response, string message)
+            : base(CreateMessage(response, message))
         {
             Response = response;
         }
 
-        public DirectoryOperationException(DirectoryResponse response, string message, Exception inner) : base(message, inner)
+        public DirectoryOperationException(DirectoryResponse response, string message, Exception inner)
+            : base(CreateMessage(response, message), inner)
         {
             Response = response;
         }
 
         public DirectoryResponse Response { get; internal set; }
+
+        private static string CreateMessage(DirectoryResponse response, string message)
+        {
+            var sb = new StringBuilder();
+            sb.Append(message ?? SR.DefaultOperationsError);
+            if (!string.IsNullOrEmpty(response?.ErrorMessage))
+            {
+                sb.Append(' ');
+                sb.Append(response.ErrorMessage);
+            }
+            return sb.ToString();
+        }
```

Comment:
This doesn't need a string builder (not that perf matters here, but that'll actually be worse).  It can instead be the simpler:
```C#
string result = message ?? SR.DefaultOperationsError;
if (!string.IsNullOrEmpty(response?.ErrorMessage))
{
    result += " " + response.ErrorMessage;
}
return result;
```

Reply:
Ah leftovers from appending the server error.